platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/maplayout/VisibilityChangeActivity.java

```diff
@@ -0,0 +1,126 @@
+package com.mapbox.mapboxsdk.testapp.activity.maplayout;
+
+import android.os.Bundle;
+import android.os.Handler;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory;
+import com.mapbox.mapboxsdk.geometry.LatLng;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.testapp.R;
+
+/**
+ * Test activity showcasing visibility changes to the mapview.
+ */
+public class VisibilityChangeActivity extends AppCompatActivity {
+
+  private MapView mapView;
+  private MapboxMap mapboxMap;
+  private Handler handler = new Handler();
+  private Runnable runnable;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_map_visibility);
+
+    mapView = (MapView) findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(MapboxMap map) {
+        mapboxMap = map;
+        mapboxMap.animateCamera(CameraUpdateFactory.newLatLngZoom(
+          new LatLng(55.754020, 37.620948), 12), 9000);
+      }
+    });
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+    handler.post(runnable = new VisibilityRunner(mapView, findViewById(R.id.viewParent), handler));
+  }
+
+  @Override
+  protected void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  private static class VisibilityRunner implements Runnable {
+
+    private MapView mapView;
+    private View viewParent;
+    private Handler handler;
+    private int counter;
+
+    VisibilityRunner(MapView mapView, View viewParent, Handler handler) {
+      this.mapView = mapView;
+      this.viewParent = viewParent;
+      this.handler = handler;
+    }
+
+    @Override
+    public void run() {
+      if (mapView != null && viewParent != null) {
```

Comment:
What about simplifying this chain of `if else`? Expressions are hard to read/follow.